pipelines/ApPipe.yaml

```diff
@@ -49,31 +51,32 @@ subsets:
       - transformDiaSrcCat
       - diaPipe
     description: >
       An alias of ApPipe to use in higher-level pipelines.
 contracts:
   # DiaPipelineTask needs diaSource fluxes, catalogs, warped exposures, and difference exposures
-  - imageDifference.doMeasurement is True
-  - imageDifference.doWriteSources is True
-  - imageDifference.doWriteWarpedExp is True
-  - imageDifference.doWriteSubtractedExp is True
-  - imageDifference.doSkySources == transformDiaSrcCat.doRemoveSkySources
+  - detectAndMeasure.doSkySources == transformDiaSrcCat.doRemoveSkySources
   # Inputs and outputs must match
   # Use of ConnectionsClass for templated fields is a workaround for DM-30210
-  - retrieveTemplate.connections.ConnectionsClass(config=retrieveTemplate).outputExposure.name ==
-      imageDifference.connections.ConnectionsClass(config=imageDifference).inputTemplate.name
-  - imageDifference.connections.ConnectionsClass(config=imageDifference).subtractedExposure.name ==
+  - retrieveTemplate.connections.ConnectionsClass(config=retrieveTemplate).template.name ==
+      subtractImages.connections.ConnectionsClass(config=subtractImages).template.name
+  - detectAndMeasure.connections.ConnectionsClass(config=detectAndMeasure).subtractedMeasuredExposure.name ==
       transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).diffIm.name
-  - imageDifference.connections.ConnectionsClass(config=imageDifference).diaSources.name ==
+  - detectAndMeasure.connections.ConnectionsClass(config=detectAndMeasure).diaSources.name ==
       transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).diaSourceCat.name
-  - imageDifference.connections.ConnectionsClass(config=imageDifference).subtractedExposure.name ==
+  - detectAndMeasure.connections.ConnectionsClass(config=detectAndMeasure).subtractedMeasuredExposure.name ==
         diaPipe.connections.ConnectionsClass(config=diaPipe).diffIm.name
-  - imageDifference.connections.ConnectionsClass(config=imageDifference).inputTemplate.name ==
+  - subtractImages.connections.ConnectionsClass(config=subtractImages).template.name ==
         diaPipe.connections.ConnectionsClass(config=diaPipe).template.name
-  - imageDifference.connections.ConnectionsClass(config=imageDifference).exposure.name ==
+  - subtractImages.connections.ConnectionsClass(config=subtractImages).science.name ==
       diaPipe.connections.ConnectionsClass(config=diaPipe).exposure.name
   - transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).diaSourceTable.name ==
         diaPipe.connections.ConnectionsClass(config=diaPipe).diaSourceTable.name
+  - subtractImages.connections.ConnectionsClass(config=subtractImages).difference.name ==
+      detectAndMeasure.connections.ConnectionsClass(config=detectAndMeasure).difference.name
+  - subtractImages.connections.ConnectionsClass(config=subtractImages).science.name ==
+      detectAndMeasure.connections.ConnectionsClass(config=detectAndMeasure).science.name
```

Comment:
I've tried to keep the contracts in dependency order, to make them easier to find in the massive wall of code. Could this block go between new lines 66 and 67?

Reply:
Though it looks like the split into `subtractImages` and `detectAndMeasure` broke the ordering anyway. Would you be willing to have a second commit that rearranges the contracts so that the `subtractImages` ones go before the `detectAndMeasure` ones?